content/en/_index.md

```diff
@@ -212,10 +218,77 @@ requests or if they are thread-safe).
 {{< /tab >}}
 {{< /tabpane >}}
 
+### Cyclomatic Complexity
+
+Cyclomatic complexity is the measure of possible code paths. The more code
+paths, the more complext the code snippet, and the hard to understand and test.
+The presences of conditionals (if/else) is an exmaple of cyclomatic complexicy.
+Code snippets should always have a single path demonstrating it's purpose with
+no extranous code.
+
+### Error Handling
+
+Samples should include examples and details of how to catch and handle common
+errors that are the result of improper interactions with the client or service. 
+
+If there is no solution (or if the solution is too verbose to resolve in a
+sample). it is accetable ot either log or leave a comment clearly explaining how
+the user should correct the situation.
+
+{{< tabpane langEqualsHeader=true >}}
+ {{< tab header="Java" >}}
+    // Follow the Google Java style guide and catch the most specific type of Exception, instead of a more general one.
+    try {
+      // Do something
+    } catch (IllegalArgumentException ok) {
+      // IllegalArgumentException's are thrown when an invalid argument has been passed to a function. Ok to ignore.
+    }
+{{< /tab >}}
+{{< /tabpane >}}
+
+### Linting
+
+Our code snippets should adhere toa. stlye used by the language communities. We
+should prefer using standard linters that are popular in the community to
+enforce style.
+
+### Portability
+
+Each of the major operating systems are well-represented among GCP users. To
+provide a good experience for most GCP users, our code snippets need to work on
+multiple operating systems. In addition, our code snippets may be executed in
+multiple hosting environments, such as GCF, GAE, GKE, GCE, AWS, and local
+machines.
+
+Where possible, code snippets should work across environments and hosting
+platforms. Code snippets should point out any differences between platforms via
+code comments and use platform-agnostic libraries and code constructs.
+
+For example, a code snippet that:
+
+* Uses the language’s standard path library to construct file paths
+* Does not directly rely on POSIX system calls
+* Handles code-sensitivity differences
+* Avoids platform-specific APIs
+
 ## Testing
 
-// TODO
+### Coverage
+Code snippets should have reasonable test coverage and all critical code paths
+should have integration tests that test against the production service. 
```

Comment:
Not sure it fits here, I think every sample should be expected to have at least one integration or end-to-end test. The language here leaves a lot of room.

Reply:
It feels like a minimum of one test would be required for a "critical code path". Do you want that stated more clearly or is there something else you want?